### ISSUE TITLE
Show URL in server error messages

### ISSUE THREAD

**Issue by axcore** [blackjack4494/yt-dlc#155]
- [x ] I'm reporting a feature request
- [x ] I've verified that I'm running youtube-dlc version **2020.10.31**
- [x ] I've searched the bugtracker for similar feature requests including closed ones

[Tartube](https://github.com/axcore/tartube) is a GUI for youtube-dl(c). It has two modes of operation: download videos, or fetch a list of videos.

When we fetch a list of videos, Tartube adds the option `--dump-json`. youtube-dl(c) dumps the JSON to STDOUT, and Tartube interprets it.

Here is the problem. When we fetch a list of videos from a channel or playlist, it is impossible to know which video produces the error messages. So Tartube is forced to mark the whole channel or playlist as _broken_, rather than marking just one video. This is very confusing for users.

I think there are three solutions:

1. Do nothing, Tartube remains broken forever
2. When the extractor raises an exception, show the URL in STDOUT. This is a one-line fix in YoutubeDL.extract_info:

```except ExtractorError as e:  # An error we somewhat expected
    self.report_error('%s: %s' % (url, compat_str(e)), e.format_traceback())
```

The before and after error messages:

`ERROR: This video requires payment to watch.`
`ERROR: https://www.youtube.com/watch?v=jXdnHUwIMs4: This video requires payment to watch.`

3. Add a new command-line option, perhaps something like `--dump-error-url`. When an extractor exception is raised, the URL is not added to the self.report_error() call, unless  `--dump-error-url` was specified.

If you think #3 is a good solution, then I can provide code in a pull request.

**Comment by blackjack4494** [comment]
Yes you are right. Someone already requested this and I had some solution for it but simply forgot it somehow (lots of things were going on at once back then).

**Comment by axcore** [comment]
Some error messages already include the video ID, but many don't. I could work with either the ID or the URL, or both.
